azerothcore/azerothcore-wotlk
# fix(Core/GameObject): Make fishing bobber finish a channeled spell rather than interrupt in all cases.

## Changes Proposed:
- Removes a scenario in which a gameobject is interacted with prematurely, where it then interrupts a spell rather than finishing it.

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/3701
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/12260

## SOURCE:

Wrath Classic PTR.

## Tests Performed:

- Tested in-game.

## How to Test the Changes:

1. `.learn 7620`
2. `.additem 6256`
3. Go to a fishing spot.
4. Fish, click the bobber when it catches a fish, before it's caught a fish, let it time out, move after casting, un-equip the pole after casting, do whatever you want. The spell channel bar should *never* remain after you've cancelled it.

## Known Issues and TODO List:


- Could not get `player->InterruptSpell` to actually disable the channel no matter what overloads I provided to it. Don't know if the function is just borked or if it doesn't apply to channeled spells, or what. The end visual result to the player is the same as when the spell is finished, i.e. when it actually has caught a fish on Classic, so the same behavior is likely fine here.
- ~~`tmpfish` is used in another gameobject interaction case, goobers, but I have no idea what that's for. I left it untouched there, but that's the reason I didn't gut the variable entirely.~~ I gutted the variable entirely.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
